common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolverSuite.java

```diff
@@ -145,29 +143,4 @@ public void jsonSerializationOfExecutorRegistration() throws IOException {
     assertEquals(shuffleInfo, mapper.readValue(legacyShuffleJson, ExecutorShuffleInfo.class));
   }
 
-  @Test
-  public void testNormalizeAndInternPathname() {
```

Comment:
Hi, @attilapiros . Could you explain why we need to remove the existing test coverage in this `Improvement` PR?

Reply:
@dongjoon-hyun sure, here you are:

The `createNormalizedInternedPathname` was as good as it was close to `java.io.FileSystem#normalize()` as in the old code the String interning could only save as any memory when its result was in equal with `java.io.FileSystem#normalize()` which was called within `File` constructor. If there was any difference in the string then the path in `File` would use a different (not interned) string as that string would be transformed a bit more.

When the test was created in the first place if they could call `java.io.FileSystem#normalize()` somehow within the tests only then they would given asserts where the expected result of `createNormalizedInternedPathname` would be `java.io.FileSystem#normalize()` (instead of the hardcoded string paths).  The test should check whether the same transformation is done on the incoming path depending on the OS. 

So now as we can call indirectly the `java.io.FileSystem#normalize()` we could rewrite the old test but that would mean having assert where `java.io.FileSystem#normalize()` is checked whether it is really `java.io.FileSystem#normalize()`. This would be a trivial assert as it is always true (like an `assertTrue(true)` just longer). So this is why we do not need the old tests.

Reply:
So the test existed because Spark has been dealing with normalization by itself (`createNormalizedInternedPathname`), despite the fact we know File object will do the normalization. Given we get rid of the custom normalization, the test would become whether normalization in JDK File object works properly or not, say, testing JDK functionality, which feels redundant.

If we cannot rely on the JDK implementation then createNormalizedInternedPathname should be just rewritten to the optimized one and this test would then keep as it is, but I'm afraid it's good direction we don't trust JDK implementation.

Reply:
>  but I'm afraid it's good direction we don't trust JDK implementation.

Let's assume JDK introduces a problem then the path is not totally normalized but still that string is interned when you use this PR so you saved the butes. Your normalized path could be even better `java.io.FileSystem#normalize()` still the sole purpose of the `createNormalizedInternedPathname` method is to save heap.

Regarding memory saving you are as good as close to get to `java.io.FileSystem#normalize()`.

Reply:
Here are the JDK tests for normalized paths:
- For Windows: https://github.com/openjdk/jdk/blob/6bab0f539fba8fb441697846347597b4a0ade428/test/jdk/java/nio/file/Path/PathOps.java#L1292-L1380
- For Unix: https://github.com/openjdk/jdk/blob/6bab0f539fba8fb441697846347597b4a0ade428/test/jdk/java/nio/file/Path/PathOps.java#L1986-L2018

Reply:
My bad “if” was missed between “direction” and “we”. Sorry about that. I made clear I have some sort of belief for JDK implementation and the maintenance, otherwise I would suggest to just port the optimized code into here. That said, to avoid further miscommunication, I’m positive on removing test, and I said it even before.

Reply:
@dongjoon-hyun Are you OK with the answer? If you're OK with it I'll move this forward.

Reply:
Yep~